### PR TITLE
feat: add `navigation` method

### PR DESCRIPTION
**Pull request by KrisQ**
## Why are these changes introduced?

Addresses [ENG-8323](https://nacelle.atlassian.net/browse/ENG-8323) 

## What is this pull request doing?

- Add the `navigation` method to the commerce queries plugin

### Unit
- On the monorepo root `npm run bootstrap`
- Navigate to `packages/storefront-sdk-plugins/commerce-queries` 
- `npm run test` every test should pass

### Local
- In the plugin directory
- - `npm run build`
- In a local project `package.json` dependencies, add the local path to the `commerce-queries` plugin
```
"@nacelle/storefront-sdk-plugin-commerce-queries": "file: LOCAL_PATH"
```
- - `npm install`
- - In a JS/TS file, add the following code:
```js
import {StorefrontClient} from '@nacelle/storefront-sdk'
import commerceQueriesPlugin from '@nacelle/storefront-sdk-plugin-commerce-queries'

const storefrontEndpoint = "https://storefront.api.development.nacelle.com/graphql/v1/spaces/SPACE_ID"
const ClientWithCommerceQueries = commerceQueriesPlugin(StorefrontClient);
const client = new ClientWithCommerceQueries({ storefrontEndpoint });

client.navigation({groupId: 'main-menu'}).then(res => console.log(res)).catch(err => console.log(err))
```
- - Make sure the code works as expected:
Screen Shot 2023-01-25 at 9 47 35 AM

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [x] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).
- [x] Updated the `README.md` with documentation changes (if applicable).

[ENG-8323]: https://nacelle.atlassian.net/browse/ENG-8323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/silver-dolphins-promise.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@nacelle/storefront-sdk-plugin-commerce-queries': patch
+---
+
+Add `navigation` method to `@nacelle/storefront-sdk-plugin-commerce-queries`: which fetches navigation data from a Nacelle Space.
```
Comment (NWRichmond) [inline .changeset/silver-dolphins-promise.md:5]: The fact that it's a change to @nacelle/storefront-sdk-plugin-commerce-queries is already accounted for in the changeset, so we can simplify the language here:

```suggestion
Add `navigation` method to fetch navigation data from a Nacelle space.
```

**Review by NWRichmond — APPROVED** [review]
As in https://github.com/getnacelle/nacelle-js/pull/317#pullrequestreview-1270185031, everything is working as expected in local testing:

@nacelle:storefront-sdk-plugin-commerce-queries demo navigation

and it would be nice to try to reduce the `applyAfter` type assertion in a future PR.

Thanks for this PR, @KrisQ!

**Review by mdarrik — APPROVED** [review]
LGTM! Works as expected! Thanks Kristenn

**Review by NWRichmond — APPROVED** [review]
Thanks for that update, @KrisQ! I'm seeing the expected type of `{ data }` returned by the `navigation` method now:

navigation data has expected type

**Pull request merged**